service-bus.api.md

```diff
@@ -0,0 +1,337 @@
+    receiveMode: "peekLock",
+    options?: GetReceiverOptions
+  ): Receiver<ReceivedMessageWithLock>;
+  getDeadLetterReceiver(
+    topicName: string,
+    subscriptionName: string,
+    receiveMode: "receiveAndDelete",
+    options?: GetReceiverOptions
+  ): Receiver<ReceivedMessage>;
+  getReceiver(
+    queueName: string,
+    receiveMode: "peekLock",
+    options?: GetReceiverOptions
+  ): Receiver<ReceivedMessageWithLock>;
+  getReceiver(
+    queueName: string,
+    receiveMode: "receiveAndDelete",
+    options?: GetReceiverOptions
+  ): Receiver<ReceivedMessage>;
+  getReceiver(
+    topicName: string,
+    subscriptionName: string,
+    receiveMode: "peekLock",
+    options?: GetReceiverOptions
+  ): Receiver<ReceivedMessageWithLock>;
+  getReceiver(
+    topicName: string,
+    subscriptionName: string,
+    receiveMode: "receiveAndDelete",
+    options?: GetReceiverOptions
+  ): Receiver<ReceivedMessage>;
+  getSender(queueOrTopicName: string, options?: GetSenderOptions): Sender;
+  getSessionReceiver(
+    queueName: string,
+    receiveMode: "peekLock",
+    options?: GetSessionReceiverOptions
+  ): SessionReceiver<ReceivedMessageWithLock>;
+  getSessionReceiver(
+    queueName: string,
+    receiveMode: "receiveAndDelete",
+    options?: GetSessionReceiverOptions
+  ): SessionReceiver<ReceivedMessage>;
+  getSessionReceiver(
+    topicName: string,
+    subscriptionName: string,
+    receiveMode: "peekLock",
+    options?: GetSessionReceiverOptions
+  ): SessionReceiver<ReceivedMessageWithLock>;
+  getSessionReceiver(
+    topicName: string,
+    subscriptionName: string,
+    receiveMode: "receiveAndDelete",
+    options?: GetSessionReceiverOptions
+  ): SessionReceiver<ReceivedMessage>;
+  getSubscriptionRuleManager(
+    topic: string,
+    subscription: string,
+    options?: GetSubscriptionRuleManagerOptions
+  ): SubscriptionRuleManager;
+}
+
+// @public
+export interface ServiceBusClientOptions {
+  retryOptions?: RetryOptions;
+  webSocketOptions?: WebSocketOptions;
+}
+
+// @public
+export interface ServiceBusMessage {
+  body: any;
+  contentType?: string;
+  correlationId?: string | number | Buffer;
+  label?: string;
+  messageId?: string | number | Buffer;
+  partitionKey?: string;
+  replyTo?: string;
+  replyToSessionId?: string;
+  scheduledEnqueueTimeUtc?: Date;
+  sessionId?: string;
+  timeToLive?: number;
+  to?: string;
+  userProperties?: {
+    [key: string]: any;
+  };
+  viaPartitionKey?: string;
+}
+
+// @public
+export interface ServiceBusMessageBatch {
+  readonly count: number;
+  readonly maxSizeInBytes: number;
+  readonly sizeInBytes: number;
+  tryAdd(message: ServiceBusMessage): boolean;
+}
+
+// @public
+export interface SessionMessageHandlerOptions {
+  autoComplete?: boolean;
+  maxConcurrentCalls?: number;
+}
+
+// @public
+export interface SessionReceiver<
+  ReceivedMessageT extends ReceivedMessage | ReceivedMessageWithLock
+> extends Receiver<ReceivedMessageT> {
+  diagnostics: {
```

Comment:
`diagnostics` can give the impression that one can have diagnostic operations on the state of the receiver. If the intention here is to avoid confusion between `peek()` the method and `peeklock` the mode, then we can experiment with `browse` first